### PR TITLE
SAA-554 commit to introduce finer grained feature switching for outgoing events.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
All (current) outgoing events and now feature switched individually.

### CODE REVIEW

File: helm_deploy/hmpps-activities-management-api/values.yaml [hunk 1]
```diff
@@ -37,6 +37,9 @@ generic-service:
     SCHEDULE_AHEAD_DAYS_ONLINE: 14
     JMS_LISTENER_MODE: "ON"
     FEATURE_EVENTS_SNS_ENABLED: false
+    FEATURE_EVENT_ACTIVITIES_ACTIVITY-SCHEDULE_CREATED: true
+    FEATURE_EVENT_ACTIVITIES_PRISONER_ALLOCATED: true
+    FEATURE_EVENT_APPOINTMENTS_APPOINTMENT-INSTANCE_CREATED: true
     FEATURE_AUDIT_SERVICE_HMPPS_ENABLED: false
     FEATURE_AUDIT_SERVICE_LOCAL_ENABLED: true
```
Comment (timharrison-moj) [inline helm_deploy/hmpps-activities-management-api/values.yaml:44]: Should HMPPS audit be true by default (unless it's not ready yet?)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventsService.kt [hunk 2]
```diff
@@ -21,20 +22,24 @@ class InboundEventsService {
...
+        ACTIVITY_SCHEDULE_CREATED -> publisher.send(ACTIVITY_SCHEDULE_CREATED.event(ScheduleCreatedInformation(identifier)))
+        PRISONER_ALLOCATED -> publisher.send(PRISONER_ALLOCATED.event(PrisonerAllocatedInformation(identifier)))
+        APPOINTMENT_INSTANCE_CREATED -> publisher.send(
+          APPOINTMENT_INSTANCE_CREATED.event(
+            AppointmentInstanceCreatedInformation(identifier),
+          ),
+        )
+      }
+    } else {
+      log.info("Ignoring publishing of event type $outboundEvent")
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventsService.kt:42]: I like it!

**Review by timharrison-moj — APPROVED** [review]
LGTM

File: helm_deploy/hmpps-activities-management-api/values.yaml (hunk 1 above)
Comment (matt-crowson-moj) [reply helm_deploy/hmpps-activities-management-api/values.yaml:44]: Not yet no.

We are still waiting on the cloud platform changes so we can read the necessary kube env vars and also add them here.  I believe @sioldham  is looking at this.

**Pull request merged**